plugins/aws/resoto_plugin_aws/resource/sagemaker.py

```diff
@@ -0,0 +1,4682 @@
+        "candidate_status": S("CandidateStatus"),
+        "inference_containers": S("InferenceContainers", default=[])
+        >> ForallBend(AwsSagemakerAutoMLContainerDefinition.mapping),
+        "creation_time": S("CreationTime"),
+        "end_time": S("EndTime"),
+        "last_modified_time": S("LastModifiedTime"),
+        "failure_reason": S("FailureReason"),
+        "candidate_properties": S("CandidateProperties") >> Bend(AwsSagemakerCandidateProperties.mapping),
+    }
+    candidate_name: Optional[str] = field(default=None)
+    final_auto_ml_job_objective_metric: Optional[AwsSagemakerFinalAutoMLJobObjectiveMetric] = field(default=None)
+    objective_status: Optional[str] = field(default=None)
+    candidate_steps: List[AwsSagemakerAutoMLCandidateStep] = field(factory=list)
+    candidate_status: Optional[str] = field(default=None)
+    inference_containers: List[AwsSagemakerAutoMLContainerDefinition] = field(factory=list)
+    creation_time: Optional[datetime] = field(default=None)
+    end_time: Optional[datetime] = field(default=None)
+    last_modified_time: Optional[datetime] = field(default=None)
+    failure_reason: Optional[str] = field(default=None)
+    candidate_properties: Optional[AwsSagemakerCandidateProperties] = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AwsSagemakerAutoMLJobArtifacts:
+    kind: ClassVar[str] = "aws_sagemaker_auto_ml_job_artifacts"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "candidate_definition_notebook_location": S("CandidateDefinitionNotebookLocation"),
+        "data_exploration_notebook_location": S("DataExplorationNotebookLocation"),
+    }
+    candidate_definition_notebook_location: Optional[str] = field(default=None)
+    data_exploration_notebook_location: Optional[str] = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AwsSagemakerResolvedAttributes:
+    kind: ClassVar[str] = "aws_sagemaker_resolved_attributes"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "auto_ml_job_objective": S("AutoMLJobObjective", "MetricName"),
+        "problem_type": S("ProblemType"),
+        "completion_criteria": S("CompletionCriteria") >> Bend(AwsSagemakerAutoMLJobCompletionCriteria.mapping),
+    }
+    auto_ml_job_objective: Optional[str] = field(default=None)
+    problem_type: Optional[str] = field(default=None)
+    completion_criteria: Optional[AwsSagemakerAutoMLJobCompletionCriteria] = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AwsSagemakerModelDeployConfig:
+    kind: ClassVar[str] = "aws_sagemaker_model_deploy_config"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "auto_generate_endpoint_name": S("AutoGenerateEndpointName"),
+        "endpoint_name": S("EndpointName"),
+    }
+    auto_generate_endpoint_name: Optional[bool] = field(default=None)
+    endpoint_name: Optional[str] = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AwsSagemakerAutoMLJob(AwsResource):
```

Comment:
Would it make sense to define an abstract `AwsSagemakerJob` class? 
A couple of attributes are shared between all jobs.